program/include/rcmail_action.php

```diff
@@ -151,6 +151,26 @@ public static function table_output($attrib, $table_data, $show_cols, $id_col)
         return $table->show($attrib);
     }
 
+    /**
+     * Set env-flag for quota. It is used in templates to determine if quota
+     * data should be displayed.
+     * In case the quota is zero and the config option
+     * `quota_zero_as_unlimited` is true, then the account has no quota and
+     * should not be shown a meaningless (or even wrong) quota-display.
+     */
+    public static function quota_set_env()
+    {
+        $rcmail = rcmail::get_instance();
+        // This call also checks the QUOTA capability of the server, we don't
+        // need to do that manually (the return value is `false` if the server
+        // doesn't support it).
+        $quota = $rcmail->storage->get_quota();
+        $zero_as_unlimited = (bool) $rcmail->config->get('quota_zero_as_unlimited');
+        // Show the quota display only if it's not unlimited.
+        $show_quota = is_array($quota) && !($quota['total'] === 0 && $zero_as_unlimited);
```

Comment:
It's not working for me since the $quota array is empty in my case. 

Something similar to this could work.

```suggestion
        $show_quota = empty($quota) || (is_array($quota) && !($quota['total'] === 0 && $zero_as_unlimited);
```

Reply:
I had considered that we could also skip displaying the quota-element if there's no quota data to show. That goes a little beyond the original issue's content but seemed right to me, because currently in this case (empty quota array) the element is shown as if no quota were used at all and its title is "unknown", which to me feels like a rather bad fallback.

(At least that's the case in my setup. Is it the same in yours?)

I'll happily use your suggestion if you still think it's right, just wanted to explain my change.

Reply:
It's empty for me - it still works and shows "unlimited".